addons/survey/tests/test_certification_flow.py

```diff
@@ -146,3 +148,76 @@ def test_flow_certificate(self):
+            certification = self.env['survey.survey'].create({
+                'title': 'User randomized Certification',
+                'questions_layout': 'page_per_section',
+                'questions_selection': 'random',
+                'state': 'open',
+            })
+
+            page1 = self._add_question(
+                None, 'Page 1', None,
+                sequence=1,
+                survey_id=certification.id,
+                is_page=True,
+                random_questions_count=1,
+            )
+
+            q101 = self._add_question(
+                None, 'What is the answer to the first question?', 'simple_choice',
+                sequence=2,
+                constr_mandatory=True, constr_error_msg='Please select an answer', survey_id=certification.id,
+                labels=[
+                    {'value': 'The correct answer', 'is_correct': True, 'answer_score': 1.0},
+                    {'value': 'The wrong answer'},
+                ])
+
+            q102 = self._add_question(
+                None, 'What is the answer to the second question?', 'simple_choice',
+                sequence=3,
+                constr_mandatory=True, constr_error_msg='Please select an answer', survey_id=certification.id,
+                labels=[
+                    {'value': 'The correct answer', 'is_correct': True, 'answer_score': 1.0},
+                    {'value': 'The wrong answer'},
+                ])
+
+        # Step: employee takes the randomized certification
+        # --------------------------------------------------
+        self.authenticate('user_emp', 'user_emp')
+
+        # Employee opens start page
+        response = self._access_start(certification)
+
+        # -> this should have generated a new user_input with a token
+        user_inputs = self.env['survey.user_input'].search([('survey_id', '=', certification.id)])
+        self.assertEqual(len(user_inputs), 1)
+        self.assertEqual(user_inputs.partner_id, self.user_emp.partner_id)
+        answer_token = user_inputs.token
+
+        # Employee begins survey with first page
+        response = self._access_page(certification, answer_token)
+        self.assertResponse(response, 200)
+        csrf_token = self._find_csrf_token(response.text)
+
+        with patch.object(IrMailServer, 'connect'):
+            question_ids = user_inputs.question_ids
+            self.assertEqual(len(question_ids), 1, 'Only one question should be selected')
+            # Whatever which question was selected, the correct answer is the first one
+            self._answer_question(question_ids, question_ids.labels_ids.ids[0], answer_token, csrf_token)
+
+        user_inputs.invalidate_cache()
+
+        answers_correctness = certification._get_answers_correctness(user_inputs)
+        self.assertEqual(answers_correctness[user_inputs], {
+            'correct': 1,
+            'skipped': 0,
+            'incorrect': 0,
+            'partial': 0,
+        })
```

Comment:
Maybe add an explicit assertion failure message?

Suggestion:
"With the configured randomization, there should be exactly 1 correctly answered question (and none skipped)."

Reply:
Seems reasonable.